src/header.rs

```diff
@@ -183,15 +186,18 @@ fn extract_full_array_shape(mut dtype: &DType) -> (Vec<u64>, &DType) {
 pub(crate) fn convert_value_to_shape(field: &Value) -> io::Result<Vec<u64>> {
     if let Value::List(ref lengths) = *field {
         lengths.iter().map(convert_value_to_shape_integer).collect()
+    } else if let Value::Tuple(ref lengths) = *field {
```

Comment:
Minor nit, I would either change this (and the other places where `else if let Value::Tuple` was added) to a `match` or, even more preferrably, since this logic occurs in many places, I would suggest factoring it out into a private `fn convert_value_to_sequence(&Value) -> &[Value]`